src/Form/Formbot.js

```diff
@@ -110,15 +114,27 @@ export default class Formbot extends React.Component {
 
   validateField(field) {
     return new Promise(resolve => {
-      const validation = this.props.validations[field];
+      const fieldState = this.state.fields[field] || {};
+      if (fieldState.validated) {
+        resolve();
+        return;
+      }
 
-      if (!validation) return;
+      const fromSchema = !!this.props.validationSchema;
+      const validation = (fromSchema ? this.props.validationSchema : this.props.validations)[field];
```

Comment:
Could these two lines be condensed into `(this.props.validationSchema || this.props.validations)[field]`?